### ISSUE TITLE
Headscale logs bearer tokens

### ISSUE THREAD

**Issue by apollo13** [juanfont/headscale#1259]
**Bug description**

Looking at the headscale logs, it logs this at info level when accessing the HTTP api:
```
2023-03-11T21:13:56Z INF unary dur=0.815623 md={":authority":"/var/run/headscale.sock","authorization":"Bearer XXX.XXX","content-type":"application/grpc","grpcgateway-accept":"*/*","grpcgateway-authorization":"Bearer XXX.XXX","grpcgateway-user-agent":"python-httpx/0.23.3","user-agent":"grpc-go/1.51.0","x-forwarded-for":"xxxxx","x-forwarded-host":"xxxxxx"} method=ListApiKeys req={} service=headscale.v1.HeadscaleService
```

This includes the whole bearer token. It would be great if the credentials wouldn't get logged :)

**Comment by disconn3ct** [comment]
Is that the official response from the project to this security report? I am more than willing to get it a CVE if that helps.

**Comment by apollo13** [comment]
Hi @disconn3ct, also not a maintainer but please don't try to get a CVE. You are apparently operating under the assumption that a project has to live up to certain security standards. This is not necessarily the case (noone what the security conduct for headscale is) and no one forces you to use headscale. Threatening to issue a CVE will usually only have the effect of getting completely ignored.

If you want to invest your time productively, you could check if this issue is actually still an issue (I don't know I don't use headscale anymore) and if yes try to come up with a patch.

**Comment by disconn3ct** [comment]
> also not a maintainer

and yet here you are, LARPing as one. Maybe you should let the actual maintainers have a chance to respond.

**Comment by disconn3ct** [comment]
@juanfont fyi https://www.cve.org/CVERecord?id=CVE-2023-47390 was assigned

**Comment by masterwishx** [comment]
using v 0.22.3 i have a lot of this when moved to goodieshq/headscale-admin from headscale-ui

**Comment by disconn3ct** [comment]
Same here. @kradalby can you please reopen this? Or would you prefer a new bug (and new CVE)

`headscale-59b8474d86-q55x6 headscale 2024-02-29T14:24:43Z INF unary dur=28.159111 md={":authority":"/var/run/headscale/headscale.sock","authorization":"Bearer OHNO_A_BEARER_TOKEN","content-type":"application/grpc","grpcgateway-accept":"application/json","grpcgateway-accept-language":"en-US,en;q=0.5","grpcgateway-authorization":"Bearer  OHNO_ANOTHER!_BEARER_TOKEN","grpcgateway-referer":"https://MY.SITE/admin/nodes/","grpcgateway-user-agent":"Mozilla/5.0 (X11; Linux x86_64; rv:122.0) Gecko/20100101 Firefox/122.0","user-agent":"grpc-go/1.54.0","x-forwarded-for":"10.108.0.5, 10.244.2.36","x-forwarded-host":"MY.SITE"} method=ListUsers req={} service=headscale.v1.HeadscaleService`

**Comment by apollo13** [comment]
Folks, read the ticket history. This is fixed and will be part of 0.23.0

**Comment by masterwishx** [comment]
> Folks, read the ticket history. This is fixed and will be part of 0.23.0

Sure, I saw it's fixed in latest prerelease fow now... Sorry for this
